File: apps/electron/src/renderer/lua/player.lua
player = {}

---@return number|nil
function player.id()
	local ok, out = pcall(swf.UserID)
	if not ok then return nil end

	return json.parse(out)
end

---@return table<table>|nil
function player.factions()
	local ok, out = pcall(swf.GetFactions)
	if not ok then return nil end

	return json.parse(out)
end

---@return string|nil
function player.username()
	local ok, out = pcall(swf.GetUsername)
	if not ok then return nil end

	return json.parse(out)
end

---@return string|nil
function player.password()
	local ok, out = pcall(swf.GetPassword)
	if not ok then return nil end

	return json.parse(out)
end

---@return string|nil
function player.class()
	local ok, out = pcall(swf.Class)
	if not ok then return nil end

	return json.parse(out)
end

---@return number|nil
function player.state()
	local ok, out = pcall(swf.State)
	if not ok then return nil end

	return json.parse(out)
end

---@return number|nil
function player.hp()
	local ok, out = pcall(swf.Health)
	if not ok then return nil end

	return json.parse(out)
end

---@return number|nil
function player.max_hp()
	local ok, out = pcall(swf.HealthMax)
	if not ok then return nil end

	return json.parse(out)
end

---@return number|nil
function player.mp()
	local ok, out = pcall(swf.Mana)
	if not ok then return nil end

	return json.parse(out)
end

---@return number|nil
function player.max_mp()
	local ok, out = pcall(swf.ManaMax)
	if not ok then return nil end

	return json.parse(out)
end

---@return number|nil
function player.level()
	local ok, out = pcall(swf.Level)
	if not ok then return nil end

	return json.parse(out)
end

---@return number|nil
function player.gold()
	local ok, out = pcall(swf.Gold)
	if not ok then return nil end

	return json.parse(out)
end

---@return boolean|nil
function player.is_afk()
	local ok, out = pcall(swf.IsAfk)
	if not ok then return nil end

	return util.str_to_bool(out)
end

---@return boolean|nil
function player.is_member()
	local ok, out = pcall(swf.IsMember)
	if not ok then return nil end

	return util.str_to_bool(out)
end

File: apps/electron/src/renderer/player.js

class Player {
	static get bank() {
		return Bank;
	}

	static get inventory() {
		return Inventory;
	}

	static get house() {
		return House;
	}

	static get tempInventory() {
		return TempInventory;
	}

	static get factions() {
		return Flash.call(window.swf.GetFactions);
	}

	static get quests() {
		return Quests;
	}

	static get className() {
		return Flash.call(window.swf.Class);
	}

	static get state() {
		return Flash.call(window.swf.State);
	}

	static get hp() {
		return Flash.call(window.swf.Health);
	}

	static get maxHp() {
		return Flash.call(window.swf.HealthMax);
	}

	static get isAlive() {
		return Player.hp > 0;
	}

	static get mp() {
		return Flash.call(window.swf.Mana);
	}

	static get maxMp() {
		return Flash.call(window.swf.ManaMax);
	}

	static get level() {
		return Flash.call(window.swf.Level);
	}

	static get gold() {
		return Flash.call(window.swf.Gold);
	}

	static get isAfk() {
		return Flash.call(window.swf.IsAfk);
	}

	static get isMember() {
		return Flash.call(window.swf.IsMember);
	}
}